core/src/main/java/org/quiltmc/mapping/entry/info/ArgEntry.java

```diff
@@ -38,7 +39,15 @@ public static ArgEntry parse(QuiltMappingParser parser) {
 		while (parser.hasValues()) {
 			String name = parser.valueName();
 			switch (name) {
-				case "index" -> index = parser.integerValue();
+				case "index" -> {
+					// validation: index cannot be negative
+					int value = parser.integerValue();
+					if (value < 0) {
+						throw new InvalidValueException(parser, "argument index cannot be negative");
+					}
+
+					index = value;
+				}
```

Comment:
Ah i didnt notice this eariler, but this is checked on line 56, so the extra check here isnt needed.

Reply:
I noticed that after implementing the check and it just doesn't seem to work
I'll check again but I remember testing and it didn't catch my number